core/src/main/java/io/confluent/kafka/schemaregistry/storage/MasterElector.java

```diff
@@ -0,0 +1,29 @@
+/**
+ * Copyright 2017 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package io.confluent.kafka.schemaregistry.storage;
+
+import io.confluent.kafka.schemaregistry.exceptions.SchemaRegistryStoreException;
+import io.confluent.kafka.schemaregistry.exceptions.SchemaRegistryTimeoutException;
+
+public interface MasterElector {
+
+  void init() throws SchemaRegistryTimeoutException, SchemaRegistryStoreException;
+
+  SchemaIdRange nextRange() throws SchemaRegistryStoreException;
```

Comment:
Yeah. I don't really like the name/conflation of interfaces. The thing that really throws me is `MasterElector` naming, which is functionally what it does, but then nothing in the interface refers to that since master elections are all implicit via initial init and callbacks to `setMaster`. A name like `Coordinator` would feel a bit cleaner and reasonably cover both aspects of the functionality, but that specific name is already overloaded in Kafka.